### PR TITLE
fix osx test

### PR DESCRIPTION
**Pull request by wxchan**

### CODE REVIEW

**Comment by wxchan** [comment]
@guolinke I don't know why but this does help pass the tests.

File: .travis/setup.sh [hunk 1]
```diff
@@ -1,7 +1,8 @@
 #!/bin/bash
 
 if [[ $TRAVIS_OS_NAME == "osx" ]]; then
-    brew install openmpi  # use openmpi to install gcc
+    brew install gcc --without-multilib
```
Comment (henry0312) [inline .travis/setup.sh]: `--without-multilib` may be not needed for tests

Comment (guolinke) [reply .travis/setup.sh]: @henry0312 I will merge this first. you can test the one without ```--without-multilib``` later.

**Comment by wxchan** [comment]
eh...look like it's not the right way to fix it.

The error msg is `/usr/local/Homebrew/Library/Homebrew/brew.rb:12:in `<main>': Homebrew must be run under Ruby 2.3! (RuntimeError)`. So it skips the first `brew install` and run the second.

But it fixes the issue temporarily.

**Comment by wxchan** [comment]
There is a warning at the beginning, I think we should fix here. (but I am not sure how now.)

```
Fix WWDRCA Certificate

$ rvm use

Warning! PATH is not properly set up, '/Users/travis/.rvm/gems/ruby-2.0.0-p648/bin' is not at first place,

         usually this is caused by shell initialization files - check them for 'PATH=...' entries,

         it might also help to re-add RVM to your dotfiles: 'rvm get stable --auto-dotfiles',

         to fix temporarily in this shell session run: 'rvm use ruby-2.0.0-p648'.

Using /Users/travis/.rvm/gems/ruby-2.0.0-p648
```

**Comment by StrikerRUS** [comment]
@wxchan The warning at the beginning of the log was presented in builds without errors: 8 days ago, for [example](https://travis-ci.org/Microsoft/LightGBM/jobs/281798162#L15).

**Comment by StrikerRUS** [comment]
I suppose we should monitor this for the answer: https://github.com/travis-ci/travis-ci/issues/8552

**Comment by StrikerRUS** [comment]
I've tried the proposed workaround `export HOMEBREW_NO_AUTO_UPDATE=1`. In this case `openmpi` installs without error, but `gcc` still could not be found https://travis-ci.org/StrikerRUS/LightGBM/jobs/285512280#L53.

**Comment by StrikerRUS** [comment]
`brew update` solves the problem: https://travis-ci.org/StrikerRUS/LightGBM/jobs/285941919

**Comment by StrikerRUS** [comment]
Seems that there are no other solutions: https://github.com/Homebrew/brew/issues/3299#issuecomment-335870756.
We could either wait the update of Travis OSX images, or perform it by ourselves by `brew update`.

**Comment by StrikerRUS** [comment]
https://blog.travis-ci.com/2017-10-16-a-new-default-os-x-image-is-coming

OK, waiting for 15 November...

**Pull request merged**